File: Programmers/level2/가장 큰 정사각형 찾기.cpp
#include <iostream>
#include<vector>
using namespace std;

int solution(vector<vector<int>> board)
{
    int answer = 0;

    if(board.size() == 1 && board[0].size() == 1)
        return (board[0][0]);
    for(int i=1; i<board.size(); i++)
    {
        for(int j=1; j<board[0].size(); j++)
        {
            if(board[i][j] == 0) continue;
            board[i][j] = min(board[i-1][j], min(board[i-1][j-1], board[i][j-1])) + 1;
            if(answer < board[i][j])
                answer = board[i][j];
        }
    }
    answer *= answer;
    return answer;
}

File: Programmers/Java/level2/가장 큰 정사각형 찾기.java
class Solution
{
    public int solution(int [][]board)
    {
        int answer = 0;

        if(board.length == 1 && board[0].length == 1)
            return (board[0][0]);
        for(int i=1; i<board.length; i++)
        {
            for(int j=1; j<board[0].length; j++)
            {
                if(board[i][j] == 0) continue;
                board[i][j] = 1 + 
                    Math.min(board[i-1][j], Math.min(board[i][j-1], board[i-1][j-1]));
                if(board[i][j] > answer)
                    answer = board[i][j];
            }
        }
        answer *= answer;
        return answer;
    }
}
